internal/credentials/kubernetes/ecr/pod_identity.go

```diff
@@ -38,20 +38,32 @@ type podIdentityCredentialHelper struct {
 func NewPodIdentityCredentialHelper(ctx context.Context) credentials.Helper {
 	logger := logging.LoggerFromContext(ctx)
 	var awsAccountID string
-	if os.Getenv("AWS_CONTAINER_CREDENTIALS_FULL_URI") == "" {
-		logger.Info("AWS_CONTAINER_CREDENTIALS_FULL_URI not set; assuming EKS Pod Identity is not in use")
+	usesIRSA := os.Getenv("AWS_ROLE_ARN") != "" && os.Getenv("AWS_WEB_IDENTITY_TOKEN_FILE") != ""
+	if os.Getenv("AWS_CONTAINER_CREDENTIALS_FULL_URI") != "" {
+		logger.Info("EKS Pod Identity appears to be in use")
+	} else if usesIRSA {
+		logger.Info("AWS_WEB_IDENTITY_TOKEN_FILE and AWS_ROLE_ARN set; assuming IRSA is being used")
+	} else {
+		logger.Info("Neither AWS_CONTAINER_CREDENTIALS_FULL_URI nor AWS_WEB_IDENTITY_TOKEN_FILE and AWS_ROLE_ARN are set; assuming neither EKS Pod Identity nor IRSA are in use")
```

Comment:
This logic appears more complex than it needs to be. Logic-wise, _everything else_ in this credential helper doesn't actually care if you're using Pod Identity or using IRSA. To the Go code, they're the same. So the logic here could really be as simple as "if at least one of these three is set, we're using managed identity of some kind and we're not otherwise."

This also makes me realize this helper should probably be renamed since it will pull double-duty for Pod Identity _and_ IRSA. `managedIdentityCredentialHelper` seems as if it would sufficiently cover both scenarios.

Reply:
> Logic-wise, everything else in this credential helper doesn't actually care if you're using Pod Identity or using IRSA. 

That is true, but IMO there should at least be something that indicates whether IRSA or EKS Pod Identity is being used, to aid people in debugging in case something goes wrong.

Reply:
Fair. Log it once up front depending on which env var you find. Beyond that, I wouldn't put additional branches in the logic.

We should still rename the component.